tests/main/dbus-activation-name-conflict/task.yaml

```diff
@@ -0,0 +1,31 @@
+summary: Test handling of D-Bus service name conflicts
+
+prepare: |
+    snap set system experimental.user-daemons=true
+    snap set system experimental.dbus-activation=true
+
+restore: |
+    snap unset system experimental.dbus-activation
+    snap unset system experimental.user-daemons
+
+execute: |
+    if ! tests.session has-system-systemd-and-dbus; then
+        echo "System does not have a systemd managed D-Bus system bus"
+        exit 0
+    fi
+
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+
+    echo "Install a snap that provides the io.snapcraft.SnapDbusService session service"
+    snap install --edge test-snapd-dbus-service
+    [ -f /var/lib/snapd/dbus-1/services/io.snapcraft.SnapDbusService.service ]
+
+    echo "Installing a second snap that tries to provide the name fails"
+    if install_local test-snapd-dbus-service-conflicting; then
```

Comment:
should this check that the error is about what we expect and not something else unrelated?

Reply:
I've updated the test to check that the error message is about the name conflict.